src/coreclr/jit/emit.h

```diff
@@ -1796,12 +1798,23 @@ class emitter
             return _idEvexNdContext != 0;
         }
 
+        bool idIsEvexZuContextSet() const
+        {
+            return (_idEvexZuContext != 0) && (IsSETZUccInstruction(_idIns));
```

Comment:
I think we want to assert this is only being called for `SETZUccInstruction` rather than check it.

We generally shouldn't be querying this unless we know it could be used.

Reply:
I think inversely we'd want to assert `idIsEvexNdContextSet` isn't being called for a `SETZUCC` instruction